pkg/logs/client/http/destination.go

```diff
@@ -32,41 +34,54 @@ var (
 	errClient = errors.New("client error")
 	errServer = errors.New("server error")
 	tlmSend   = telemetry.NewCounter("logs_client_http_destination", "send", []string{"endpoint_host", "error"}, "Payloads sent")
+
+	expVarIdleMsMapKey = "idleMs"
+	expVarInUseMapKey  = "inUseMs"
 )
 
 // emptyPayload is an empty payload used to check HTTP connectivity without sending logs.
-var emptyPayload []byte
+var emptyPayload = message.Payload{}
 
 // Destination sends a payload over HTTP.
 type Destination struct {
+	sync.Mutex
 	url                 string
 	apiKey              string
 	contentType         string
 	host                string
-	contentEncoding     ContentEncoding
 	client              *httputils.ResetClient
 	destinationsContext *client.DestinationsContext
-	once                sync.Once
-	payloadChan         chan []byte
 	climit              chan struct{} // semaphore for limiting concurrent background sends
+	wg                  sync.WaitGroup
 	backoff             backoff.Policy
 	nbErrors            int
 	blockedUntil        time.Time
 	protocol            config.IntakeProtocol
 	origin              config.IntakeOrigin
+	lastError           error
+	isRetrying          bool
+	shouldRetry         bool
+	expVars             *expvar.Map
```

Comment:
This list of fields in `Destination struct` could use some ordering - either functional or alphabetic. As-is, it's hard to parse all of the important ones.